src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
```

Comment:
The function lacks type safety as it assumes that the `key` property of `T` is always a string. Consider adding a constraint on `T` to ensure that `key` is a property with string values.